### PR TITLE
[Beam-115] More complete translation of the graph through the Runner API

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @vikkyrk

**Comment by robertwb** [comment]
retest this please

**Comment by robertwb** [comment]
Apache Beam :: SDKs :: Java :: IO :: MongoDB broken, retest this please

File: sdks/python/apache_beam/io/iobase.py [hunk 1]
```diff
@@ -812,6 +819,23 @@ def display_data(self):
                                       label='Read Source'),
             'source_dd': self.source}
 
+  def to_runner_api_parameter(self, context):
+    # Avoid circular imports.
+    from apache_beam.runners.api import beam_runner_api_pb2
```
Comment (vikkyrk) [inline sdks/python/apache_beam/io/iobase.py]: Any reason not to move the registration out of this method?

File: sdks/python/apache_beam/utils/urns.py [hunk 2]
```diff
@@ -24,3 +33,90 @@
...
 
 PICKLED_TRANSFORM = "beam:ptransform:pickled_python:v0.1"
+FLATTEN_TRANSFORM = "beam:ptransform:flatten:v0.1"
+WINDOW_INTO_TRANSFORM = "beam:ptransform:window_into:v0.1"
+READ_TRANSFORM = "beam:ptransform:read:v0.1"
+
+PICKLED_SOURCE = "beam:source:pickled_python:v0.1"
+
+
+class RunnerApiFn(object):
```
Comment (vikkyrk) [inline sdks/python/apache_beam/utils/urns.py]: +1 to this registration util function.

File: sdks/python/apache_beam/io/iobase.py (hunk 1 above)
Comment (robertwb) [inline sdks/python/apache_beam/io/iobase.py]: Circular import issues...

**Comment by robertwb** [comment]
Looks like the tests are failing because I'm now requiring Sources to be pickle-able, which the CreateSource is not. Will fix.

**Comment by vikkyrk** [comment]
The create source was able to be serialized by the `dill` the last time I checked.

**Comment by robertwb** [comment]
I've been pretty busy with other stuff, so simply removed the source changes for now to be taken up in a future CL (either myself or someone else). PTAL.

**Comment by vikkyrk** [comment]
LGTM (modulo jenkins tests passing)

**Comment by robertwb** [comment]
Jenkins https://builds.apache.org/job/beam_PreCommit_Java_MavenInstall/10069 and travis (Python) are both happy.

**Pull request closed without merging**